### ISSUE TITLE
`Counter.inc()` and `Counter.dec()` should have keyword arguments

### ISSUE THREAD

**Issue by garysassano** [winglang/wing#4059]
### I tried this:

```
bring cloud;

let counter = new cloud.Counter(initial: 100);

test "key inc" {
  let k = "foo";
  counter.inc(key: k);
}
```

### This happened:

```sh
|   counter.inc(key: k);
|   ^^^^^^^^^^^^^^^^^^^ No named arguments expected
```

### I expected this:

I expected that I could call `Counter.inc()` just with the name of the key, without specifying the amount, since by default it should know that I want to increment by 1.

### Component

SDK

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by hasanaburayyan** [comment]
I think the change seems like a sound DX improvement.

Some additional conversation was had here: https://winglang.slack.com/archives/C048QCN2XLJ/p1693763658564429?thread_ts=1693732987.636579&cid=C048QCN2XLJ

**Comment by Chriscbr** [comment]
Makes sense -- `counter.inc(nil, "key")` is a bit of an API design smell to me. (iI both arguments to the function are optional, I don't see a need to require folks to specify them in a predefined order)

**Comment by garysassano** [comment]
Keep.

**Comment by ekeren** [comment]
It seems to be working:
```wing
bring cloud; 
bring expect;

let counter = new cloud.Counter();

test "main" {
  counter.inc(1, "k1");
  counter.inc(10, "k2");
  expect.equal(1, counter.peek("k1"));
  expect.equal(10, counter.peek("k2"));
}
```



[playground](https://www.winglang.io/play/?code=YgByAGkAbgBnACAAYwBsAG8AdQBkADsAIAAKAGIAcgBpAG4AZwAgAGUAeABwAGUAYwB0ADsACgAKAGwAZQB0ACAAYwBvAHUAbgB0AGUAcgAgAD0AIABuAGUAdwAgAGMAbABvAHUAZAAuAEMAbwB1AG4AdABlAHIAKAApADsACgAKAHQAZQBzAHQAIAAiAG0AYQBpAG4AIgAgAHsACgAgACAAYwBvAHUAbgB0AGUAcgAuAGkAbgBjACgAMQAsACAAIgBrADEAIgApADsACgAgACAAYwBvAHUAbgB0AGUAcgAuAGkAbgBjACgAMQAwACwAIAAiAGsAMgAiACkAOwAKACAAIABlAHgAcABlAGMAdAAuAGUAcQB1AGEAbAAoADEALAAgAGMAbwB1AG4AdABlAHIALgBwAGUAZQBrACgAIgBrADEAIgApACkAOwAKACAAIABlAHgAcABlAGMAdAAuAGUAcQB1AGEAbAAoADEAMAAsACAAYwBvAHUAbgB0AGUAcgAuAHAAZQBlAGsAKAAiAGsAMgAiACkAKQA7AAoAfQA%3D)

**Comment by garysassano** [comment]
@ekeren Not working: [Playground](https://www.winglang.io/play/?code=YgByAGkAbgBnACAAYwBsAG8AdQBkADsAIAAKAGIAcgBpAG4AZwAgAGUAeABwAGUAYwB0ADsACgAKAGwAZQB0ACAAYwBvAHUAbgB0AGUAcgAgAD0AIABuAGUAdwAgAGMAbABvAHUAZAAuAEMAbwB1AG4AdABlAHIAKAApADsACgAKAHQAZQBzAHQAIAAiAG0AYQBpAG4AIgAgAHsACgAgACAAbABlAHQAIABrACAAPQAgACIAZgBvAG8AIgA7AAoAIAAgAGMAbwB1AG4AdABlAHIALgBpAG4AYwAoAGsAZQB5ADoAIABrACkAOwAKAH0A)

**Comment by garysassano** [comment]
@ekeren The point of the issue was to avoid having to do things like `counter.inc(1, "k1");` from your example.

**Comment by ekeren** [comment]
I see. I miss read it 

💯

**Comment by Chriscbr** [comment]
To recap, I think this is what the proposed API would look like:

```js
// OK
counter.inc(key: "foo");
counter.inc(amount: 5);
counter.inc(key: "foo", amount: 5);
counter.inc(amount: 5, key: "foo");

// invalid
counter.inc(5);
counter.inc("foo");
```

The alternative options:
- Keep the existing API, where the amount is the first positional argument and the key is the second positional argument
- Additively make this change via a new method like `counter.update(...)` and leave the existing methods
